velox/core/Expressions.h

```diff
@@ -136,14 +136,19 @@ class ConstantTypedExpr : public ITypedExpr {
       return false;
     }
 
+    if (!this->type()->kindEquals(casted->type())) {
```

Comment:
"There is still a problem where constants of complex types may be reported "equal" to similar constants of built in types. "

kindEquals actually check each child's type? It can't handle that case? Thanks!

Reply:
This problem concerns complex types that use custom types, e.g. MAP(BIGINT(), JSON()). This type is considered same as MAP(BIGINT(), VARCHAR()).